indico/modules/events/registration/models/registrations.py

```diff
@@ -229,9 +229,15 @@ class Registration(db.Model):
     )
     #: Type of consent given to publish this registration
     consent_to_publish = db.Column(
-        PyIntEnum(PublishConsentType),
+        PyIntEnum(RegistrationVisibility),
         nullable=False,
-        default=PublishConsentType.nobody
+        default=RegistrationVisibility.nobody
+    )
+    #: Management-set override for visibility
+    visibility_override = db.Column(
+        PyIntEnum(RegistrationVisibility),
```

Comment:
I wouldn't make this an enum but rather a simple `participant_hidden` boolean. By making this an enum you're adding a huge amount of extra complexity which would otherwise be as simple as adding a check for `and not participant_hidden` to the queries etc...

The point of the feature is simply that if a participant asks the organizers to not be shown in the participant list, or the organizer wishes to hide someone (for whatever reason), they have the ability to do so, even in an event that's not consent-based.